File: Bookworm Backend Dot net/Models/Customer.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using System;
using Microsoft.EntityFrameworkCore;

namespace Bookworm.Models
{
    [Table("customer_master")]
    [Index(nameof(Email), IsUnique = true)]
    [Index(nameof(Phone), IsUnique = true)]
    public class Customer
    {
        [Key]
        [Column("customer_id")]
        public int Id { get; set; }

        [StringLength(255)]
        [Column("address")]
        public string? Address { get; set; }

        [Column("age")]
        public int? Age { get; set; }

        [Column("dob", TypeName = "date")]
        public DateTime? DateOfBirth { get; set; }

        [Required]
        [StringLength(100)]
        [Column("email")]
        public string Email { get; set; }

        [Required]
        [StringLength(100)]
        [Column("name")]
        public string Name { get; set; }

        [Required]
        [StringLength(255)]
        [Column("password_hash")]
        public string PasswordHash { get; set; }

        [StringLength(15)]
        [Column("phone")]
        public string? Phone { get; set; }

        // Navigation property for one-to-many relationship with Invoice
        public virtual ICollection<Invoice> Invoices { get; set; } = new HashSet<Invoice>();

        // Many-to-many relationship with Role via the CustomerRole join entity
        public virtual ICollection<CustomerRole> CustomerRoles { get; set; } = new HashSet<CustomerRole>();

        // Navigation property for one-to-many relationship with Cart
        public virtual ICollection<Cart> Carts { get; set; } = new HashSet<Cart>();
    }
}

File: Bookworm Backend Spring Boot/src/main/java/com/bookworm/entities/Customer.java
package com.bookworm.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

/*
 * Represents a registered customer of the service.
 * Contains personal and contact information, and links to the customer's
 * transactional history.
 * Maps to the 'customers' table.
 */
@Entity
@Getter
@Setter
@Table(name = "customer_master")
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customer_id")
    private Integer id;

    @Column(name = "name", length = 100, nullable = false)
    private String name;

    @Column(name = "email", length = 100, nullable = false, unique = true)
    private String email;

    @Column(name = "password_hash", length = 255, nullable = false)
    private String passwordHash;

    @Column(name = "phone", length = 15, unique = true)
    private String phone;

    @Column(name = "dob")
    private LocalDate dateOfBirth;

    @Column(name = "age")
    private Integer age;

    @Column(name = "address", length = 255)
    private String address;

    @OneToMany(mappedBy = "customer")
    private Set<Cart> carts;
    
    @ManyToMany(fetch = FetchType.EAGER) // REMOVED: cascade = {CascadeType.PERSIST, CascadeType.MERGE}
    @JoinTable(
            name = "customer_roles",
            joinColumns = @JoinColumn(name = "customer_id"),
            inverseJoinColumns = @JoinColumn(name = "role_id")
    )
    private Set<Role> roles = new HashSet<>();

    //    @OneToMany(mappedBy = "customer")
    //    private Set<Invoice> invoices;
    //
    //    @OneToMany(mappedBy = "customer")
    //    private Set<RentDetail> rentDetails;
    //
    //    //@OneToOne(mappedBy = "customer", cascade = CascadeType.ALL, orphanRemoval = true)
    //    private Shelf shelf;
}
